atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java

```diff
@@ -50,4 +50,16 @@ SweepBatchWithPartitionInfo getNextBatchToSweep(ShardAndStrategy shardStrategy,
         }
         return accumulator.toSweepBatch();
     }
+
+    @Immutable
+    public interface ReadBatchingRuntimeConfig {
```

Comment:
minor nits: config objects typically have just values. I'd suggest we either pass `Supplier<RBRC>` and use some `Suppliers.transform(...)` higher up, or rename this `ReadBatchingRuntimeContext` or something like that